BoxContentSDK/BoxContentSDK/Requests/BOXFileRequest.m

```diff
@@ -65,74 +65,105 @@ - (BOXAPIOperation *)createOperation
     if (fieldString.length > 0) {
         queryParameters = @{BOXAPIParameterKeyFields:fieldString};
     }
-    
-    BOXAPIJSONOperation *JSONOperation = [self JSONOperationWithURL:URL
-                                                         HTTPMethod:BOXAPIHTTPMethodGET
-                                              queryStringParameters:queryParameters
-                                                     bodyDictionary:nil
-                                                   JSONSuccessBlock:nil
-                                                       failureBlock:nil];
+
+    BOXAPIOperation *fileOperation = nil;
+    if (self.associateId != nil) {
+        BOXAPIDataOperation *dataOperation = [self dataOperationWithURL:URL
+                                                             HTTPMethod:BOXAPIHTTPMethodGET
+                                                  queryStringParameters:queryParameters
+                                                         bodyDictionary:nil
+                                                           successBlock:nil
+                                                           failureBlock:nil
+                                                            associateId:self.associateID];
+        NSString *requestDirectory = self.requestDirectoryPath;
+        if (requestDirectory.length == 0) {
+            // Emergency fallback for robustness. The request directory path is expected to be set at request configuration time.
```

Comment:
instead of falling back, I ended up adding a check if self.requestDirectoryPath exists. If it does not exist, I allow the request to be made as non-background.